agent/rpc/peering/service.go

```diff
@@ -764,16 +764,15 @@ func (s *Server) PeeringList(ctx context.Context, req *pbpeering.PeeringListRequ
 // -- ImportedServicesCount and ExportedServicesCount
 // NOTE: we return a new peering with this additional data
 func (s *Server) reconcilePeering(peering *pbpeering.Peering) *pbpeering.Peering {
+	cp := copyPeering(peering)
 	streamState, found := s.Tracker.StreamStatus(peering.ID)
 	if !found {
 		// TODO(peering): this may be noise on non-leaders
 		s.Logger.Warn("did not find peer in stream tracker; cannot populate imported and"+
```

Comment:
This warning (and subsequent setting of data) is going to execute in many tests because of the following block of logic skipping over calls to s.establishStream() by hitting this block of code. https://github.com/hashicorp/consul/blob/v1.17.1/agent/consul/leader_peering.go#L254-L256

This was effectively a bug where the data in the state store was being written to directly on accident, because there was no copy operation prior to mutation.